home_4/src/shipper/strategy/factory/AirEastCostCalculationStrategyFactory.ts

```diff
@@ -0,0 +1,35 @@
+import { Shipment } from "../../../Shipment";
+import { Letter } from "../../../Letter";
+import { Package } from "../../../Package";
+import { Oversize } from "../../../Oversize";
+import { AirEastLetterCostCalculationStategy } from "../AirEastLetterCostCalculationStrategy";
+import { CostCalculationStrategy } from "../CostCalculationStrategy";
+import { AirEastPackageCostCalculationStategy } from "../AirEastPackageCostCalculationStrategy";
+import { AirEastOversizeCostCalculationStrategy } from "../AirEastOversizeCostCalculationStrategy";
+
+export class AirEastCostCalculationStrategyFactory {
+  private static factory: AirEastCostCalculationStrategyFactory;
+
+  private map: Map<string, CostCalculationStrategy> = new Map();
+
+  private constructor() {
+    this.map.set(typeof Letter, new AirEastLetterCostCalculationStategy());
```

Comment:
`typeof` operator on Class will return 'function':
1. these lines rewrite each other
2. it's not possible to generate keys like that